modules/core/include/opencv2/core/mat.hpp

```diff
@@ -1084,6 +1084,18 @@ class CV_EXPORTS Mat
     */
     void copyTo( OutputArray m, InputArray mask ) const;
 
+#if CV_FP16
```

Comment:
Thank you for the comment.  That's exactly the point I wanted to discuss.

1) I'll follow your suggestion to move convertFp16 as a separate function.  In that case, do you have a suggestion of which module to be added ? Does this still fit in core module ?

2) I understand the background that it'll be better to have a SW emulation, but does this mean that it's a mandatory request to implement the SW emulation version ? let me understand bit more deeper before I start coding the SW emulation.
  a) SW emulation is a must request.  Without that, this PR will be rejected
  b) to have this function in cv::Mat, it's a mandatory to have a SW emulation.  If it's implemented as a separate function out of cv::Mat, it can be under conditional compilation (in another word, comment 1) and 2) are related)
  c) something else